brick/brick.go

```diff
@@ -0,0 +1,88 @@
+package brick
+
+import (
+	"crypto/md5"
+	"fmt"
+	"path"
+	"strings"
+
+	"github.com/gluster/glusterd2/utils"
+	"github.com/gluster/glusterd2/volume"
+)
+
+const (
+	// TODO: Remove hardcoding
+	//	glusterfsd = "/usr/local/sbin/glusterfsd"
+	glusterfsd = "/usr/bin/sleep"
+)
+
+type Brick struct {
+	// Externally consumable using methods of Daemon interface
+	binarypath     string
+	args           string
+	socketfilepath string
+	pidfilepath    string
+
+	// For internal use
+	brickinfo volume.Brickinfo
```

Comment:
I'd prefer not to have separate daemon package implementing a type satisfying the `daemon.Daemon` interface for every daemon. That just adds more packages to manage.

As a guideline to follow, every daemon should have its own package, which implements a whole lot more functionality related to the daemon, as well as implementing the interface. This keeps information related to the daemon in one place, not spread around, making it easier to maintain.

If implementing the interface within the `volume` package doesn't sound right, then keep this package and move `Brickinfo` here. This would adhere to the above guideline, as a brick is a standalone daemon unit.